tests/test_qdrant_client.py

```diff
@@ -2152,36 +2160,55 @@ def test_create_payload_index(prefer_grpc):
             max_token_len=7,
             lowercase=True,
         ),
+        wait=True,
     )
 
     major, minor, patch, dev = read_version()
     if major is None or dev or (major, minor, patch) >= (1, 11, 0):
-        client.create_payload_index(COLLECTION_NAME, "uuid", models.PayloadSchemaType.UUID)
+        client.create_payload_index(
+            COLLECTION_NAME, "uuid", models.PayloadSchemaType.UUID, wait=True
+        )
 
         client.create_payload_index(
             COLLECTION_NAME,
             "keyword_parametrized",
             models.KeywordIndexParams(
                 type=models.KeywordIndexType.KEYWORD, is_tenant=False, on_disk=True
             ),
+            wait=True,
         )
+        payload_schema = client.get_collection(COLLECTION_NAME).payload_schema
+        assert payload_schema["keyword_parametrized"].params.is_tenant is False
+        assert payload_schema["keyword_parametrized"].params.on_disk is True
+
         client.create_payload_index(
             COLLECTION_NAME,
             "integer_parametrized",
             models.IntegerIndexParams(
                 type=models.IntegerIndexType.INTEGER,
                 lookup=True,
-                range=True,
+                range=False,
                 is_principal=False,
                 on_disk=True,
             ),
+            wait=True,
         )
+        if prefer_grpc:
+            rest_client = QdrantClient()
+            _ = rest_client.get_collection(COLLECTION_NAME).payload_schema
+        payload_schema = client.get_collection(COLLECTION_NAME).payload_schema
+        assert payload_schema["integer_parametrized"].params.lookup is True
+        assert payload_schema["integer_parametrized"].params.range is False
+        assert payload_schema["integer_parametrized"].params.is_principal is False
+        assert payload_schema["integer_parametrized"].params.on_disk is True
```

Comment:
Just discussed this part in chat: 

This is being asserted against the results of rest client, because this is the kind of behavior that would fail before. So this is in the spirit of a regression test.